### PR TITLE
Add docs for scaling private locations

### PR DESCRIPTION
**Pull request by andrewvc**
Obs-docs counterpart to https://github.com/elastic/beats/pull/34307 . CC @paulb-elastic . 

Fixes https://github.com/elastic/observability-docs/issues/2550

### CODE REVIEW

File: docs/en/observability/synthetics-private-location.asciidoc [hunk 1]
```diff
@@ -29,10 +29,11 @@ refer to {fleet-guide}/agent-policy.html#create-a-policy[{agent} policy].
 A private location should be set up against an agent policy that runs on a single {agent}.
 Do _not_ run the same agent policy on multiple agents being used for private locations, as you may
 end up with duplicate or missing tests. Private locations do not currently load balance tests across
-multiple {agents}.
+multiple {agents}. See <<synthetics-private-location-scaling>> for information on increasing the capacity
+within a private location.
 
-Any private location can only run two tests simultaneously.
-As a result, if more than two tests are assigned to a particular private location, there may be a small delay to run them.
+Any private location can only run two browser tests simultaneously, and as many lightweight checks as resources allow.
```
Comment (colleenmcginnis) [inline docs/en/observability/synthetics-private-location.asciidoc:35]: Use the same language you use in the section below.

```suggestion
By default private locations are configured to allow two simultaneous browser tests and an unlimited number of lightweight checks.
```

File: docs/en/observability/synthetics-private-location.asciidoc [hunk 2]
```diff
@@ -125,4 +126,23 @@ When the {agent} is running you can add a new private location in {kib}:
...
\ No newline at end of file
+TIP: You can run the same monitor both from Elastic's global managed infrastructure _and_ your private locations to compare results.
+
+[discrete]
+[[synthetics-private-location-scaling]]
+= Scaling private locations
+
+By default private locations are configured to allow two simultaneous browser tests, and an unlimited number of lightweight checks. 
+These limits can be set via the environment variables `SYNTHETICS_LIMIT_{TYPE}`, where `{TYPE}` is one of `BROWSER`, `HTTP`, `TCP`, and `ICMP` 
+for the container running the {{agent}} docker image.
```
Comment (colleenmcginnis) [inline docs/en/observability/synthetics-private-location.asciidoc:137]:
```suggestion
for the container running the {agent} docker image. 
```

File: docs/en/observability/synthetics-private-location.asciidoc [hunk 3]
```diff
@@ -125,4 +126,23 @@ When the {agent} is running you can add a new private location in {kib}:
...
+[[synthetics-private-location-scaling]]
+= Scaling private locations
+
+By default private locations are configured to allow two simultaneous browser tests, and an unlimited number of lightweight checks. 
+These limits can be set via the environment variables `SYNTHETICS_LIMIT_{TYPE}`, where `{TYPE}` is one of `BROWSER`, `HTTP`, `TCP`, and `ICMP` 
+for the container running the {{agent}} docker image. 
+
+It is critical to allocate enough memory and CPU capacity to handle configured limits. 
+A good rule of thumb is to allocate at 2 GiB of memory and 2 cores per browser instance in order to have more consistent
+performance and avoid out of memory errors. Resource requirements will vary depending on workload.
```
Comment (colleenmcginnis) [inline docs/en/observability/synthetics-private-location.asciidoc:141]: I'm not sure how well "rule of thumb" translates.

```suggestion
Start by allocating at 2 GiB of memory and two cores per browser instance to ensure consistent
performance and avoid out-of-memory errors. Then adjust as needed. Resource requirements will vary depending on workload. 
```

File: docs/en/observability/synthetics-private-location.asciidoc [hunk 4]
```diff
@@ -125,4 +126,23 @@ When the {agent} is running you can add a new private location in {kib}:
...
+It is critical to allocate enough memory and CPU capacity to handle configured limits. 
+A good rule of thumb is to allocate at 2 GiB of memory and 2 cores per browser instance in order to have more consistent
+performance and avoid out of memory errors. Resource requirements will vary depending on workload. 
+For lightweight monitors much less memory is needed. We recommend starting at 512MiB of memory and two cores for 
+lightweight checks and increasing allocated memory and CPU based on observed usage patterns.
+
+Note that these limits are for simultaneous tests, not total tests. If, for instance,
+60 browser tests were scheduled to run once per hour, with each taking exactly 1 minute to run that would fully occupy one execution slot.
+That being said, it is not a good practice to leave execution slots without extra capacity. A good starting point would be to over-allocate by
+a factor of 5. In the previous example that would mean mean allocating 5 slots.
```
Comment (colleenmcginnis) [inline docs/en/observability/synthetics-private-location.asciidoc:148]:
```suggestion
These limits are for simultaneous tests, not total tests. For example, if
60 browser tests were scheduled to run once per hour and each took 1 minute to run, that would fully occupy one execution slot.
However, it is a good practice to set up execution slots with extra capacity. A good starting point would be to over-allocate by
a factor of 5. In the previous example that would mean mean allocating 5 slots.
```

File: docs/en/observability/synthetics-private-location.asciidoc [hunk 5]
```diff
@@ -125,4 +126,23 @@ When the {agent} is running you can add a new private location in {kib}:
...
+
+By default private locations are configured to allow two simultaneous browser tests, and an unlimited number of lightweight checks. 
+These limits can be set via the environment variables `SYNTHETICS_LIMIT_{TYPE}`, where `{TYPE}` is one of `BROWSER`, `HTTP`, `TCP`, and `ICMP` 
+for the container running the {{agent}} docker image. 
+
+It is critical to allocate enough memory and CPU capacity to handle configured limits. 
+A good rule of thumb is to allocate at 2 GiB of memory and 2 cores per browser instance in order to have more consistent
+performance and avoid out of memory errors. Resource requirements will vary depending on workload. 
+For lightweight monitors much less memory is needed. We recommend starting at 512MiB of memory and two cores for 
+lightweight checks and increasing allocated memory and CPU based on observed usage patterns.
```
Comment (colleenmcginnis) [inline docs/en/observability/synthetics-private-location.asciidoc:143]: Use the same structure as above. 

```suggestion
Much less memory is needed for lightweight monitors. Start by allocating at 512MiB of memory and two cores for 
lightweight checks. Then increase allocated memory and CPU based on observed usage patterns.
```

**Comment by andrewvc** [comment]
Thanks for the insightful edits @colleenmcginnis , I've merged them all :)

File: docs/en/observability/synthetics-private-location.asciidoc [hunk 6]
```diff
@@ -125,4 +126,23 @@ When the {agent} is running you can add a new private location in {kib}:
...
+[discrete]
+[[synthetics-private-location-scaling]]
+= Scaling private locations
+
+By default private locations are configured to allow two simultaneous browser tests, and an unlimited number of lightweight checks. 
+These limits can be set via the environment variables `SYNTHETICS_LIMIT_{TYPE}`, where `{TYPE}` is one of `BROWSER`, `HTTP`, `TCP`, and `ICMP` 
+for the container running the {agent} docker image. 
+
+It is critical to allocate enough memory and CPU capacity to handle configured limits. 
+Start by allocating at 2 GiB of memory and two cores per browser instance to ensure consistent
```
Comment (paulb-elastic) [inline docs/en/observability/synthetics-private-location.asciidoc:140]: Typo?
>Start by allocating at 2 GiB of memory

Start by allocating at _least_ 2 GiB of memory

File: docs/en/observability/synthetics-private-location.asciidoc [hunk 7]
```diff
@@ -125,4 +126,23 @@ When the {agent} is running you can add a new private location in {kib}:
...
+performance and avoid out-of-memory errors. Then adjust as needed. Resource requirements will vary depending on workload. 
+Much less memory is needed for lightweight monitors. Start by allocating at 512MiB of memory and two cores for
```
Comment (paulb-elastic) [inline docs/en/observability/synthetics-private-location.asciidoc:142]: Same? (at _least_...)

File: docs/en/observability/synthetics-private-location.asciidoc [hunk 8]
```diff
@@ -125,4 +126,23 @@ When the {agent} is running you can add a new private location in {kib}:
...
+It is critical to allocate enough memory and CPU capacity to handle configured limits. 
+Start by allocating at 2 GiB of memory and two cores per browser instance to ensure consistent
+performance and avoid out-of-memory errors. Then adjust as needed. Resource requirements will vary depending on workload. 
+Much less memory is needed for lightweight monitors. Start by allocating at 512MiB of memory and two cores for 
+lightweight checks. Then increase allocated memory and CPU based on observed usage patterns.
+
+These limits are for simultaneous tests, not total tests. For example, if
+60 browser tests were scheduled to run once per hour and each took 1 minute to run, that would fully occupy one execution slot.
+However, it is a good practice to set up execution slots with extra capacity. A good starting point would be to over-allocate by
+a factor of 5. In the previous example that would mean mean allocating 5 slots.
```
Comment (paulb-elastic) [inline docs/en/observability/synthetics-private-location.asciidoc:148]: Typo _mean mean_

Comment (colleenmcginnis) [reply docs/en/observability/synthetics-private-location.asciidoc:148]:
```suggestion
a factor of 5. In the previous example that would mean allocating 5 slots.
```

File: docs/en/observability/synthetics-private-location.asciidoc (hunk 7 above)
Comment (colleenmcginnis) [reply docs/en/observability/synthetics-private-location.asciidoc:142]:
```suggestion
Much less memory is needed for lightweight monitors. Start by allocating at least 512MiB of memory and two cores for 
```

File: docs/en/observability/synthetics-private-location.asciidoc (hunk 6 above)
Comment (colleenmcginnis) [reply docs/en/observability/synthetics-private-location.asciidoc:140]:
```suggestion
Start by allocating at least 2 GiB of memory and two cores per browser instance to ensure consistent
```

File: docs/en/observability/synthetics-private-location.asciidoc (hunk 7 above)
Comment (andrewvc) [reply docs/en/observability/synthetics-private-location.asciidoc:142]: I really wonder how I made this same typo twice 🤔

**Comment by andrewvc** [comment]
@paulb-elastic @colleenmcginnis I've resolved the merge conflict and resolved all comments, thank you both for your input. LMK if anything else is needed to get to LGTM.

**Review by paulb-elastic — APPROVED** [review]
LGTM

**Review by colleenmcginnis — APPROVED** [review]
This looks good to me! I have a _really_ minor comment, but it shouldn't block this PR so I'll ✅  and you can decide if/how to address it.

It bothers me a bit that the new section is _after_ "Next steps" since that is usually used as the final section on a page to send the reader off to another page.

Screen Shot 2023-02-09 at 9 23 10 AM

I see two solutions:

1. If scaling is something that most/all users should do immediately after creating a private location, then move "Scaling private locations" before "Next steps".
2. If scaling is more advanced and is usually done separately from initial setup, then remove just the "Next steps" heading and the text for that section can just fall under "Add a private location".

**Comment by andrewvc** [comment]
@colleenmcginnis not a minor point at all, I've reordered so next steps is last. I think almost everyone should be aware of the scaling concerns, so yes, all should read it.

**Comment by justinkambic** [comment]
Skipping post-ff testing for this as copy was approved and it's been merged to o11y docs.

**Pull request merged**